### PR TITLE
fix: vertical scrolling

### LINKED ISSUE #565: Suggestion: Horizontal chat input field scrolling
A suggestion to change the way the chat input field behaves:

Currently, the input field scrolls _horizontally_ when the text exceeds the field size, making it expand _vertically_ would make it much easier to read the full text entered before sending.

Example image:

### PR DESCRIPTION
**Pull request by SputNikPlop**
This changes horizontal scrolling to vertical scrolling. Closes #565

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ androidxViewpager2 = "1.1.0-beta02"
 androidxRoom = "2.6.0"
 androidxWebkit = "1.8.0"
 composeBom = "2022.10.00"
-material = "1.11.0-beta01"
+material = "1.11.0-alpha03"
```
Comment (flex3r) [inline gradle/libs.versions.toml:31]: Why was this changed?

File: app/src/main/res/layout/rm_host_bottomsheet.xml [hunk 2]
```diff
@@ -43,7 +43,7 @@
                 android:layout_height="wrap_content"
                 android:imeOptions="actionDone"
                 android:inputType="textNoSuggestions"
-                android:singleLine="true"
+                android:textMultiLine="true"
```
Comment (flex3r) [inline app/src/main/res/layout/rm_host_bottomsheet.xml:46]: The `textMultiLine` attribute does not exist

File: app/src/main/res/layout/add_channel_dialog.xml [hunk 3]
```diff
@@ -21,7 +21,7 @@
                 android:layout_height="wrap_content"
                 android:imeOptions="actionDone|flagNoExtractUi"
                 android:inputType="text"
-                android:singleLine="true" />
+                android:singleLine="false" />
```
Comment (flex3r) [inline app/src/main/res/layout/add_channel_dialog.xml:24]: This should stay single line

File: app/src/main/res/layout/edit_dialog.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
                 android:layout_height="wrap_content"
                 android:imeOptions="actionDone|flagNoExtractUi"
                 android:inputType="text"
-                android:singleLine="true" />
+                android:singleLine="false" />
```
Comment (flex3r) [inline app/src/main/res/layout/edit_dialog.xml:28]: This should stay single line

File: app/src/main/res/layout/emote_suggestion_item.xml [hunk 5]
```diff
@@ -19,6 +19,6 @@
         android:layout_gravity="center_vertical"
         android:paddingHorizontal="8dp"
         android:paddingVertical="16dp"
-        android:singleLine="true"
+        android:singleLine="false"
```
Comment (flex3r) [inline app/src/main/res/layout/emote_suggestion_item.xml:22]: This should stay single line

File: app/src/main/res/layout/user_ignore_item.xml [hunk 6]
```diff
@@ -37,7 +37,7 @@
                     android:layout_height="wrap_content"
                     android:imeOptions="actionDone"
                     android:inputType="text"
-                    android:singleLine="true"
+                    android:singleLine="false"
```
Comment (flex3r) [inline app/src/main/res/layout/user_ignore_item.xml:40]: This should stay single line since the expected input is only a username

File: app/src/main/res/layout/user_highlight_item.xml [hunk 7]
```diff
@@ -37,7 +37,7 @@
                     android:layout_height="wrap_content"
                     android:imeOptions="actionDone"
                     android:inputType="text"
-                    android:singleLine="true"
+                    android:singleLine="false"
```
Comment (flex3r) [inline app/src/main/res/layout/user_highlight_item.xml:40]: This should stay single line since the expected input is only a username

File: app/src/main/res/layout/user_display_item.xml [hunk 8]
```diff
@@ -105,7 +105,7 @@
                     android:layout_height="wrap_content"
                     android:imeOptions="actionNext"
                     android:inputType="text"
-                    android:singleLine="true"
+                    android:singleLine="false"
```
Comment (flex3r) [inline app/src/main/res/layout/user_display_item.xml:108]: This should stay single line since the expected input is only a username alias

File: app/src/main/res/layout/user_display_item.xml [hunk 9]
```diff
@@ -38,7 +38,7 @@
...
```
Comment (flex3r) [inline app/src/main/res/layout/user_display_item.xml:41]: This should stay single line since the expected input is only a username

File: app/src/main/res/layout/tts_ignore_list_item.xml [hunk 10]
```diff
@@ -31,7 +31,7 @@
                 android:layout_height="wrap_content"
                 android:imeOptions="actionDone"
                 android:inputType="text"
-                android:singleLine="true"
+                android:singleLine="false"
```
Comment (flex3r) [inline app/src/main/res/layout/tts_ignore_list_item.xml:34]: This should stay single line since the expected input is only a username

**Review by flex3r — CHANGES_REQUESTED** [review]
I don't think we can or should just change all inputs to support multiple lines.
Please also include some screenshots for the input field that are going to change when they have multiple lines

File: gradle/libs.versions.toml (hunk 1 above)
Comment (SputNikPlop) [reply gradle/libs.versions.toml:31]: I thought this was the most current version, fixed now

**Comment by SputNikPlop** [comment]
IMG-20231106-WA0000
I think this fix should work @flex3r let me know if this isn't quite what you want.

File: app/src/main/res/layout/main_fragment.xml [hunk 11]
```diff
@@ -174,10 +174,10 @@
                 android:layout_height="wrap_content"
                 android:completionThreshold="2"
                 android:imeOptions="actionSend|flagNoExtractUi"
-                android:inputType="text|textCapSentences"
+                android:inputType="text|textMultiLine"
+                android:maxLines="6"
                 android:paddingStart="12dp"
-                android:paddingEnd="8dp"
-                android:singleLine="true" />
+                android:paddingEnd="8dp" />
```
Comment (flex3r) [inline app/src/main/res/layout/main_fragment.xml:180]: I tinkered around with some paddings and I think having an additonal `android:paddingVertical="4dp"` improves some of the overlapping with the label.

**Review by flex3r — CHANGES_REQUESTED** [review]
Looks very good 👍 
Just some minor stuff remaining

**Comment by SputNikPlop** [comment]
@flex3r Hopefully I got it right this time

**Review by flex3r — APPROVED** [review]
Looks good now, thanks for the contribution!
